src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const massiveToReturn = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete copyState[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const prop in copyState) {
+        delete copyState[prop];
+      }
+    }
```

Comment:
better to rewrite this logic using the switch/case instead of the multiple if/else